src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -158,6 +158,11 @@ public long num2ulong(long num) {
             return num;
         }
 
+        @Specialization(guards = "isRubyBignum(num)")
+        public Object num2num(DynamicObject num) {
+            throw new RaiseException(getContext().getCoreExceptions().rangeError("bignum out of range of unsigned long",this));
```

Comment:
However, 2**64-1 would fit in an unsigned long but not our Fixnum, does MRI raise in such a case?

Reply:
indeed this was not complete